### ISSUE TITLE
Support TLS for memcache

### ISSUE THREAD

**Issue by ntimo** [grafana/loki#10548]
# Problem
Currently Loki does not support TLS for memecached, this however is a critical requirement for a secure setup of Loki with memcached.

# Solution
Please add support for using TLS with memcached and Loki. 

# Additional details
The Loki docs don't mention tls support: https://grafana.com/docs/loki/latest/configure/
Mimir however supports this: https://grafana.com/docs/mimir/latest/references/configuration-parameters/#memcached

**Comment by Sheikh-Abubaker** [comment]
Hi @ntimo,

I just went through the links you mentioned above and I saw in mimir configs there is a parameter tls_enabled which defaults to false but in the loki configs there is no tls_enabled like parameter within memcached block, am I correct is the thing that you are trying to tell with this issue ?

**Comment by ntimo** [comment]
> Hi @ntimo,
> 
> I just went through the links you mentioned above and I saw in mimir configs there is a parameter tls_enabled which defaults to false but in the loki configs there is no tls_enabled like parameter within memcached block, am I correct is the thing that you are trying to tell with this issue ?

Yes exactly. Therefore it seems like its not possible to configure Loki to use a TLS connection to Memcached.

**Comment by Sheikh-Abubaker** [comment]
Hi @ntimo, I'm willing to contribute to this issue and fix it, but I am just a beginner so can you please tell me or just head me to the right direction where I can learn how to define some new configuration parameter that is not defined by default, so that I can define ```tls_enabled``` parameter for loki and eventually add it to the default configs ```loki.yaml``` to configure Loki to use a TLS connection to Memcached.

**Comment by jeroen-nijssen** [comment]
Looking at the config now.. https://grafana.com/docs/loki/latest/configure/ (V3.0.x) this is possible.
